device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -269,14 +209,6 @@ public synchronized void sendEventAsync(Message message,
                                Object callbackContext,
                                String deviceId)
     {
-        /* Codes_SRS_DEVICE_IO_21_024: [If the client is closed, the sendEventAsync shall throw an IllegalStateException.] */
-        if (this.state == IotHubClientState.CLOSED)
-        {
-            throw new IllegalStateException(
```

Comment:
What happens when transport is not opened and sendEventAsync is triggered?

Reply:
The message gets to the IotHubTransport layer where the same IllegalStateException is thrown. The net result is the same to a customer, but it allows us to just have one place where we check the state before queuing a message